### PR TITLE
Convert to Async Await, Add configurable environments, remove deprecated code

### PR DESCRIPTION
**Pull request by kylebrowning**
This PR does the following.

- Removes the custom NIO code in favor of AsyncHttpClient
- removes deprecated code
- Allows environment to be sent on configure
- Moves The bearer factory to be an actor
- removes JWTKit in favor of Swift-crypto and a smaller EDSA signature

These are all breaking changes and as such will require a major version bump

### CODE REVIEW

**Comment by kylebrowning** [comment]
@lgaches any interest I didn't break anything for you?

**Comment by kylebrowning** [comment]
@lgaches any interest in testin I didn't break anything for you?

**Comment by kylebrowning** [comment]
@lgaches, @TimAEllis, @LukasCZ any interest in testing I didn't break anything for you?

**Comment by lgaches** [comment]
Sure, I can test. Would be nice to update also the README file sample code.

**Comment by lgaches** [comment]
Sure, I can test. Would be nice to update also the README file sample code.

**Comment by kylebrowning** [comment]
Will definitely update the readme!

**Comment by lgaches** [comment]
It's working for me

File: Package.swift [hunk 1]
```diff
@@ -12,31 +12,21 @@ let package = Package(
...
     dependencies: [
-        .package(url: "https://github.com/apple/swift-nio.git", from: "2.10.1"),
-        .package(url: "https://github.com/apple/swift-nio-ssl.git", from: "2.14.0"),
-        .package(url: "https://github.com/apple/swift-nio-http2.git", from: "1.13.0"),
         .package(url: "https://github.com/apple/swift-log.git", from: "1.0.0"),
         .package(url: "https://github.com/vapor/jwt-kit.git", from: "4.0.0"),
+        .package(url: "https://github.com/swift-server/async-http-client.git", from: "1.10.0")
     ],
     targets: [
         .target(name: "APNSwiftExample", dependencies: [
```
Comment (lgaches) [inline Package.swift:20]:
```suggestion
        .executableTarget(name: "APNSwiftExample", dependencies: [
```

Comment (kylebrowning) [reply Package.swift]: Fixed, thanks

**Comment by kylebrowning** [comment]
I made just a few more changes. but mostly just hiding things from the public namespce

**Pull request merged**